src/goesproc/config.cc

```diff
@@ -186,12 +181,16 @@ bool loadHandlers(const toml::Value& v, Config& out) {
                 r = ((ti & 0xFF0000) >> 16) / 255.0;
                 g = ((ti & 0xFF00) >> 8) / 255.0;
                 b = (ti & 0xFF) / 255.0;
-              } else {
+              } else if (red && green && blue) {
```

Comment:
I'm puzzled by this one: if you specify 'color' and it isn't valid, shouldn't the error be returned always?

Reply:
No ... the point is that the user can use any of:

{ color: "#ff0000" }
{ r: 1, g: 0, b: 0 }
{ h: 0: s: 1, v: 1 }

If the first conditonal matches, local vars r, g, and b get their values from the converted hex color.  If the second one matches, the local vars get their values from the red/green/blue TOML handles, which point to "r" "g" and "b" keys in the TOML table.

Farther down, if we don't have a valid RGB color from either of the above, we look at our "h" "s" and "v" TOML handles.  If still nothing, we error out.

Does that make sense?

Reply:
Looking closer, 1 sec

Reply:
You're absolutely right, that else was in the wrong place.  DERP.  Check it now and see if the logic matches what I describe above.